### PR TITLE
docs : update filesystem usage guide

### PR DESCRIPTION
**Pull request by manoharahk**
This patch adds the smartfs filesystem usage guide on
new board.

Signed-off-by: Manohara HK <manohara.hk@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: docs/HowToUseSmartFS.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+#ifdef CONFIG_S5J_SFLASH
+        s5j_sflash_init();
+#endif
+}
+```
+### MTD Driver
+MTD driver must be enabled to access the underlying FLASH driver.
+Example driver is at os/fs/driver/mtd/mtd_progmem.c.
+
+Below configuration must enabled for using mtd driver.
```
Comment (sunghan-chang) [inline docs/HowToUseSmartFS.md]: Could you explain how to enable them through menuconfig?
For example,
```
File Systems -> xxx -> yyy to set y
```

File: docs/HowToUseSmartFS.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
+# How to use SmartFS
+
+SmartFS is FileSystem used for user partition in TizenRT.
+
+SmartFS stands for Sector Mapped Allocation for Really Tiny (SMART) flash.
+It is a filesystem that has been designed to work primary with small,
```
Comment (sunghan-chang) [inline docs/HowToUseSmartFS.md]: If you want to put these lines at different line, you should add two spaces at end of lines.
These are written in same line at markdown view, but you wrote them at different lines.

File: docs/HowToUseSmartFS.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+CONFIG_SMARTFS_BAD_SECTOR
+CONFIG_SMARTFS_DYNAMIC_HEADER
+CONFIG_SMARTFS_JOURNALING
+CONFIG_SMARTFS_SECTOR_RECOVERY
+```
+Smart FS can be mounted on smart mtd device.
+Before mounting smart device formatted to smartfs using mksmartfs tool
+Example snippet for formatting and mounting smart fs
+```
+#ifdef CONFIG_SIDK_S5JT200_AUTOMOUNT_USERFS
```
Comment (sunghan-chang) [inline docs/HowToUseSmartFS.md]: 1. Is this needed at fs guide doc?
2. I think you had better mention how to mount fs in TASH also.

File: docs/HowToUseSmartFS.md (hunk 2 above)
Comment (manoharahk) [reply docs/HowToUseSmartFS.md]: Ok, It needs to be at different line. I will modify it . Thanks

File: docs/HowToUseSmartFS.md (hunk 1 above)
Comment (manoharahk) [reply docs/HowToUseSmartFS.md]: Okay, I will explain

File: docs/HowToUseSmartFS.md (hunk 3 above)
Comment (manoharahk) [reply docs/HowToUseSmartFS.md]: 1. Do you mean only #ifdef or the whole snippet?
2. I will add, how to mount fs in TASH

Comment (sunghan-chang) [reply docs/HowToUseSmartFS.md]: 1. Yes, automount functionality.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: docs/HowToUseSmartFS.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
+# How to use SmartFS
+
+SmartFS is FileSystem used for user partition in TizenRT.
```
Comment (pradeepns85) [inline docs/HowToUseSmartFS.md]: SmartFS is a

File: docs/HowToUseSmartFS.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+SmartFS stands for Sector Mapped Allocation for Really Tiny (SMART) flash.  
+It is a filesystem that has been designed to work primary with small,
+serial NOR type flash parts that are 1M byte to 16M byte in size
```
Comment (pradeepns85) [inline docs/HowToUseSmartFS.md]: Can we mention what is the least size it supports currently ?

File: docs/HowToUseSmartFS.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+media operations such as logical sector allocation, freeing and management,
+erase block management, low-level formatting, wear leveling, etc.
+
+On top of the SMART MTD layer is the Smart Filesystem Layer.  
+The SmartFS code uses the logical sector services of the SMART MTD layer to
+provide file and directory level management, such as creating new files,
+chaining logical sectors together to create files, creating directories and
+ file / directory search and management routines.
+
+## FLASH Driver
```
Comment (pradeepns85) [inline docs/HowToUseSmartFS.md]: How about having this explanation in the reverse order. as Each layer makes use of the services provided by underlying layer

File: docs/HowToUseSmartFS.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+3. Enable CONFIG_MTD_PARTITION_NAMES  
+```
+File  Systems->Memory Technology Device (MTD) Support->Support MTD partition  naming to y
+```
+4. Enable CONFIG_MTD_PROGMEM  
+```
+File  Systems->Memory Technology Device (MTD) Support->Enable on-chip program  FLASH MTD device to y
+```
+
+Example initialization is as shown below,refer os/arch/arm/src/sidk_s5jt200/src/s5jt200_tash.c
```
Comment (pradeepns85) [inline docs/HowToUseSmartFS.md]: Example of

File: docs/HowToUseSmartFS.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+#ifdef CONFIG_S5J_SFLASH
+        s5j_sflash_init();
+#endif
+}
+```
+### MTD Driver
+MTD driver must be enabled to access the underlying FLASH driver.  
+Example driver is at os/fs/driver/mtd/mtd_progmem.c.
+
+Below configuration must enabled for using mtd driver.
```
Comment (pradeepns85) [inline docs/HowToUseSmartFS.md]: must be

File: docs/HowToUseSmartFS.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+```
+File  Systems->SMART file system
+```
+2. Set CONFIG_SMARTFS_ERASEDSTATE
+```
+File  Systems->SMART file system ->SMARTFS options->FLASH erased state  (0xff) to 0xFF
+```
+3. set CONFIG_SMARTFS_MAXNAMLEN=32
+```
+File  Systems->SMART file system ->SMARTFS options->Maximum file length  (32) to 32
```
Comment (pradeepns85) [inline docs/HowToUseSmartFS.md]: is it Maximum file length or Maximum filename length ?

File: docs/HowToUseSmartFS.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+5. CONFIG_SMARTFS_VERIFY_JOURNALING
+```
+File  Systems->SMART file system ->SMARTFS options->Enable journaling for  smartfs->verify Journal logging
+```
+6. CONFIG_SMARTFS_SECTOR_RECOVERY
+```
+File  Systems->SMART file system ->SMARTFS options-> Enable recovery of  lost sectors in File system
+```
+Smart FS can be mounted on smart mtd device.  
+Before mounting smart device formatted to smartfs using mksmartfs tool.
```
Comment (pradeepns85) [inline docs/HowToUseSmartFS.md]: Before mounting,

File: docs/HowToUseSmartFS.md (hunk 4 above)
Comment (manoharahk) [reply docs/HowToUseSmartFS.md]: Ok

File: docs/HowToUseSmartFS.md (hunk 5 above)
Comment (manoharahk) [reply docs/HowToUseSmartFS.md]: Will check

File: docs/HowToUseSmartFS.md (hunk 6 above)
Comment (manoharahk) [reply docs/HowToUseSmartFS.md]: It is written in the order of which one must be brought up first during bring up phase.

File: docs/HowToUseSmartFS.md (hunk 8 above)
Comment (manoharahk) [reply docs/HowToUseSmartFS.md]: Ok

File: docs/HowToUseSmartFS.md (hunk 7 above)
Comment (manoharahk) [reply docs/HowToUseSmartFS.md]: Ok

File: docs/HowToUseSmartFS.md (hunk 9 above)
Comment (manoharahk) [reply docs/HowToUseSmartFS.md]: Filename

File: docs/HowToUseSmartFS.md (hunk 10 above)
Comment (manoharahk) [reply docs/HowToUseSmartFS.md]: Ok

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
I think this doc has many informations like explanation of our fs, porting guide and how to use it.
Could we split them? I think what configs should be enabled is a different topic.

File: docs/HowToUseSmartFS.md (hunk 9 above)
Comment (thapav) [reply docs/HowToUseSmartFS.md]: In menuconfig option is:
 Maximum file name length
We can add name before: length  (32) to 32

**Comment by sunghan-chang** [comment]
@manoharahk Could you write what user should do simply at top of file?
It has too many things in this doc, like explanaion of smartfs.
```
1. add a flash driver on arch (link of tap)
2. add blah blah
3. enable configs
    1. mtd
    2. blah blah
```

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by Taejun-Kwon** [comment]
Smartfs is one of file system in TizenRT, why don't you add procfs and romfs here???

**Pull request merged**